objects/src/accounts/account_id.rs

```diff
@@ -134,336 +108,511 @@ impl FromStr for AccountStorageMode {
     }
 }
 
+impl From<AccountId> for AccountStorageMode {
+    fn from(id: AccountId) -> Self {
+        id.storage_mode()
+    }
+}
+
+impl From<AccountIdPrefix> for AccountStorageMode {
+    fn from(id_prefix: AccountIdPrefix) -> Self {
+        id_prefix.storage_mode()
+    }
+}
```

Comment:
This seems largely redundant? Are the function's not enough, or are these required for generics somewhere?